runtime/manager.go

```diff
@@ -55,26 +61,36 @@ func NewManagerFromReaders(configReader, genesisReader io.Reader, options ...fun
 	var genesisState = new(genesis.GenesisState)
 	parse(genesisReader, genesisState)
 
-	mgr := &Manager{
-		config:       cfg,
-		genesisState: genesisState,
-		clock:        clock.New(),
+	var mgr = new(Manager)
+	bus, err := CreateBus(mgr)
+	if err != nil {
+		log.Fatalf("[ERROR] Failed to initialize bus: %v", err)
 	}
 
+	mgr.config = cfg
+	mgr.genesisState = genesisState
+	mgr.clock = clock.New()
+	mgr.bus = bus
+
 	for _, o := range options {
 		o(mgr)
 	}
 
 	return mgr
 }
 
-func NewManager(config *configs.Config, genesisState *genesis.GenesisState, options ...func(*Manager)) *Manager {
-	mgr := &Manager{
-		config:       config,
-		genesisState: genesisState,
-		clock:        clock.New(),
+func NewManager(config *configs.Config, genesis *genesis.GenesisState, options ...func(*Manager)) *Manager {
```

Comment:
Can we reuse `NewManager` in `NewManagerFromFiles`?

Reply:
Absolutely